UKHomeOffice/sgmr-service
# BUGFIX: Merge vessel and form data so all is there for validation

### AC / Description of changes
Bug reported that since 2.1 release the validation on submission always fails due to missing pleasurecraft data.
Discovered that it is not being persisted into the session data the same way the rest of the formData is.

This page is being refactored as part of 2.2 (see #529 ) but a temporary patch to make it work for 2.1 testing I have merged the formData and voyageData for validation

- formData produces dates and times in their individual pieces (hour, minute, day, month, etc)
- voyageData produces the data that will be submitted (vessels, names, etc.)


### To Test

- create a voyage with all data
- submit the voyage
- it should submit

- create a draft voyage
- try to submit the voyage
- it should fail


### Notes

---

**Remember to merge to feature branches not `master`.**
